libs/ui/src/hooks/use_usb_drive.test.tsx

```diff
@@ -53,16 +55,19 @@ test('full lifecycle with USBControllerButton', async () => {
 
   const mockKiosk = fakeKiosk();
   mockKiosk.getUsbDriveInfo.mockResolvedValue([]);
+  const { promise: mountPromise, resolve: mountResolve } = deferred<void>();
+  mockKiosk.mountUsbDrive.mockReturnValueOnce(mountPromise);
```

Comment:
Now that state updates within `useEffect`s are batched, some loading states never actually appear on screen during tests because, without actual things to wait on, components just go from `setState('loading')` to `setState('loaded')`. When we actually want to capture these loading states it may be necessary to mock unresolved promises like so.